lambda.tf

```diff
@@ -0,0 +1,63 @@
+locals {
+  code_version  = var.autoscaler_version
+  function_name = "ec2-autoscaler-${var.worker_pool_id}"
+}
+
+resource "null_resource" "download" {
+  provisioner "local-exec" {
+    command = "${path.module}/download.sh ${local.code_version}"
+  }
+}
+
+data "archive_file" "binary" {
+  type        = "zip"
+  source_file = "${var.local_path}ec2-workerpool-autoscaler_v${var.autoscaler_version}"
+  output_path = "ec2-workerpool-autoscaler_v${var.autoscaler_version}.zip"
+  depends_on  = [ null_resource.download ]
+}
+
+resource "aws_lambda_function" "autoscaler" {
+  filename         = data.archive_file.binary.output_path
+  source_code_hash = data.archive_file.binary.output_base64sha256
+  function_name    = local.function_name
+  role             = aws_iam_role.lambda.arn
+  handler          = "ec2-workerpool-autoscaler_v${var.autoscaler_version}"
+
+  runtime                        = "go1.x"
+
+  environment {
+    variables = {
+      AUTOSCALING_GROUP_ARN         = module.asg.autoscaling_group_arn
+      AUTOSCALING_REGION            = data.aws_region.this.name
```

Comment:
Can't the autoscaler determine the region it runs in automatically?

Reply:
That was my point. There is no good reason for that, but let's avoid updating the autoscaler for now.